mapswipe_workers/mapswipe_workers/generate_stats/generate_stats.py

```diff
@@ -1,8 +1,39 @@
+import datetime as dt
+from typing import List, Optional
+
+from mapswipe_workers import auth
 from mapswipe_workers.definitions import DATA_PATH, logger
 from mapswipe_workers.generate_stats import overall_stats, project_stats
 
 
-def generate_stats(project_id_list: list):
+def get_recent_projects(hours: int = 3):
+    """Get ids for projects when results have been submitted within the last x hours."""
+    pg_db = auth.postgresDB()
+    query_insert_results = """
+        select project_id
+        from results
+        -- Using timestamp attribute here which is set for all projects
+        -- and also represents the start_time for newer projects.
+        -- "Old" projects have no start_time attribute.
+        -- There is an index defined on "timestamp".
+        where "timestamp" >= %(timestamp)s
+        group by project_id
+    """
+    timestamp = (dt.datetime.utcnow() - dt.timedelta(hours=hours)).isoformat()[
+        0:-3
+    ] + "Z"
+    project_info = pg_db.retr_query(query_insert_results, {"timestamp": timestamp})
+    del pg_db
```

Comment:
Usually it is not required to delete pg_db manually. This will be done automatically once the program leaves the scope of `get_recent_projects`.

Reply:
fixed